src/averageAges.js

```diff
@@ -37,7 +37,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const female = people.filter(person =>
+    withChildren ? person.sex === 'f' && people
+      .some(element => element.mother === person.name)
+      : person.sex === 'f');
+
+  const everage = (arr) => {
+    return arr.reduce((sum, person) =>
+      sum + (person.died - person.born), 0) / arr.length;
+  };
```

Comment:
We can move this logic to separate function and reuse it to avoid code duplication